CODING_GUIDELINES.md

```diff
@@ -6,30 +6,82 @@ This document contains the coding guidelines for the ASPIRE Python project.
-For automatic checking of style problems, [Pycharm from Jet Brains](https://www.jetbrains.com/pycharm/) 
-is a good candidate which has the professional version for academic users with free license fee. 
-
-
-## Best Practices
-
-For beginners in scientific computing, some general guides are suggested in Greg Wilson's papers as below:
-
- 1. [Good Enough Practices in Scientific Computing]( https://doi.org/10.1371/journal.pcbi.1005510) 
- 2. [Best Practices for Scientific Computing]( https://doi.org/10.1371/journal.pbio.1001745) 
+and [PEP 257 Docstring Conventions](https://www.python.org/dev/peps/pep-0257/) guidelines. [This webpage](https://realpython.com/documenting-python-code/) is a good guide for documenting Python code.
+For an editor with some automatic checking of style problems, [Pycharm from Jet Brains](https://www.jetbrains.com/pycharm/)
+is a good candidate which has the professional version for academic users with free license fee.
 
+In the future compoenents of PEP8 and PEP257 will be checked programmatically.  Documentation of a standard tool to perform
+these operations locally will be provided to developers at that time.
 
 ## Source Code Control
 
-As an open source software, we use Git and GitHub for our source code control. 
-The following links are useful tutorials. 
+As an open source software, we use Git with GitHub for our source code control.
+The following links are useful tutorials.
 1. Git tutorials from [Atalassian](https://www.atlassian.com/git/tutorials) and [Git-SCM](https://git-scm.com/docs/gittutorial)
-2. GitHub guides from [GitHub](https://guides.github.com/) 
+2. GitHub guides from [GitHub](https://guides.github.com/)
 
-## Git Workflow Changes
+## Git Workflow
 
-Our project follows [Gitflow Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).
+Our project follows a reduced [Gitflow Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).
 Please submit any PRs against the `develop` branch.
 
 ![Gitflow Diagram](https://wac-cdn.atlassian.com/dam/jcr:61ccc620-5249-4338-be66-94d563f2843c/05%20(2).svg?cdnVersion=357)
 
+The basic idea is that we have `feature` branches,  `develop` as an integration focused branch, and `master` for releases.
+We do not currently have a need for a dedicated release or staging branch; in our case `develop` also serves this purpose.
+
+Developers are welcome to maintain their own fork, but this is not required.
+
+When `develop` is at a milestone where we want to cut a release we will commit with a PEP440 compliant `Major.Minor` version
+scheme, tag it similarly, and complete a merge to master following the controls below. It is at this point the
+release would be a candadite to upload to PyPI.
+
+###  Git Workflow Controls
+
+To facilitate effective integration, merges into `develop` must occur as a pull request that is approved by two developers.
```

Comment:
Two reviewers?

Reply:
Yes, that is what we talked about, in at least three meetings now, including only a few hours ago? This is a major component of how we avoid conflicts. If you review a code and it is going to conflict with your work that the other developer isn't privy to yet, we discuss those areas during the review, maybe associating with area of concern in the other work. It also ensures certain architectural concerns aren't overlooked because some code gets smashed in quickly.

Reply:
I means probably we should replace two developers with two reviewers since Joakim's role is defined as a codeowner.

Reply:
Cool, sure will change.